### PR TITLE
Bugfix - Teams map

### PR DESCRIPTION
**Pull request by GeniJaho**
https://trello.com/c/ig4zVBBQ

Trying to fix the teams' map by removing the existing code that uses the leaflet marker cluster library, and replacing it with a simplified version of Supercluster called `ClusterMap.vue`.
The map is now hidden by default, and it's shown when you select a team from the dropdown, time filters do not apply.
Added a new database table called `team_clusters`, and a new generator command `php artisan clusters:generate-team-clusters`, scheduled to run every Monday at 00:20.
Refactored some code related to the GlobalMapController and GlobalClustersController, so that it's also used on the TeamClusterController.

Haven't modified the Supercluster.vue at all to keep the global map safe from breaking changes. I simply made a smaller copy of it, just to have it working.

The teams' maps are now working correctly, I do however get a console error sometimes when I zoom to the points level, and zoom out and in a few times, and then click on the map. The issue is caused by the Leaflet library unable to delete all its event listeners when we clear/remove the map when changing teams. There were no good solutions for that through the Github issues, but since maps are not used much and most users belong to only one team, this shouldn't be a big issue.

### CODE REVIEW

File: resources/js/components/Teams/TeamMap.vue [hunk 1]
```diff
@@ -1,109 +1,69 @@
...
-            );
-        },
-
         fullscreenChange (fullscreen)
         {
-            this.fullscreen = fullscreen
+            this.fullscreen = fullscreen;
         },
 
         toggle ()
```
Comment (xlcrr) [inline resources/js/components/Teams/TeamMap.vue:52]: Comments needed on every function and computed property

**Review by xlcrr — APPROVED** [review]

File: app/Traits/GeohashTrait.php [hunk 2]
```diff
@@ -86,8 +86,6 @@ private function calculateAdjacent($srcHash, $dir)
      */
     public function neighbors ($srcHash)
     {
-        $geohashPrefix = substr($srcHash, 0, strlen($srcHash) - 1);
```
[the remark this answers is not in the document]
Comment (GeniJaho) [reply app/Traits/GeohashTrait.php:89]: The variable wasn't used anywhere in the method.

File: config/app.php [hunk 3]
```diff
@@ -123,8 +123,6 @@
 
     'cipher' => 'AES-256-CBC',
 
-    'root_dir' => env('APP_ROOT_DIR', '/home/forge/openlittermap.com'),
```
[the remark this answers is not in the document]
Comment (GeniJaho) [reply config/app.php:126]: Yes, I have added that .env variable only for this command.

File: resources/js/components/Teams/TeamMap.vue (hunk 1 above)
Comment (GeniJaho) [reply resources/js/components/Teams/TeamMap.vue:52]: This code is basically copy-pasted from the User profile map. 👍

**Pull request merged**